transit_odp/publish/forms/seasonal_services.py

```diff
@@ -190,3 +198,72 @@ def clean(self):
         if start > end:
             raise ValidationError("Start date must be earlier than end date")
         return cleaned_data
+
+
+class SeasonalServiceEditDateForm(GOVUKModelForm):
+    class Meta:
+        model = SeasonalService
+        fields = (
+            "start",
+            "end",
+        )
+
+    start = forms.DateTimeField(
+        required=True,
+        label=_("Service begins on"),
+        error_messages={
+            "invalid": _("Error first date"),
+            "required": _("This date is required"),
+        },
+        widget=NumberInput(attrs={"type": "date"}),
+    )
+
+    end = forms.DateTimeField(
+        required=True,
+        label=_("Service ends on"),
+        error_messages={
+            "invalid": _("Error last date"),
+            "required": _("This date is required"),
+        },
+        widget=NumberInput(attrs={"type": "date"}),
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.org_id = kwargs.pop("org_id", None)
+        self.page = kwargs.pop("page", None)
+        super().__init__(*args, **kwargs)
+
+    def get_layout(self):
+        help_modal = render_to_string(
+            "publish/snippets/help_modals/seasonal_services.html"
+        )
+        return Layout(
+            HTML(
+                format_html(
+                    '<h2 class="govuk-heading-s">Service operating dates {}</h2>',
+                    help_modal,
+                ),
+            ),
+            DateDiv(
+                FieldNoErrors("start", wrapper_class="date"),
+                FieldNoErrors("end", wrapper_class="date"),
+                css_class="date-container",
+            ),
+            SEPARATOR,
+            ButtonHolder(
+                SAVE_BUTTON,
+                cancel_seasonal_service(self.org_id, self.page),
+                css_class="buttons",
+            ),
+        )
+
+    def clean(self):
```

Comment:
This is different than SeasonalServiceEditRegistrationCodeDateForm.clean I think we this should be inherited

Reply:
No it's exactly the same, I don't know what you mean, the only thing that I can tell you now is if I inherit from 
`SeasonalServiceEditRegistrationCodeDateForm` I inherit also the 

`
transit_odp/publish/forms/seasonal_services.py:176->186

    def clean_registration_code(self):
        registration_code = self.cleaned_data["registration_code"]
        exists = SeasonalService.objects.filter(
            licence=self.licence, registration_code=registration_code
        ).exists()
        if exists:
            raise ValidationError(
                "This service code has already been set up with a date range"
            )
        return registration_code
`  

and I don't want this in the Edit View